MapboxGeocoder/MBGeocoder.swift

```diff
@@ -346,6 +356,114 @@ open class Geocoder: NSObject {
         components.queryItems = params
         return components.url!
     }
+    
+    /**
+     Send a keydown event
+    */
+    func initKeydown(options: GeocodeOptions){
+        if (options.queries.count > 1) {
+            // don't record batch geocodes
+            return
+        }
+        let query = options.queries[0]
+        if (isReverseGeocode(testStr: query)){
+            // don't record reverse geocodes
+            return
+        }
+        if (query == self.lastQuery){
+            // don't duplicate
+            return
+        }
+        if (query == ""){
+            // don't record empty queries
+            return
+        }
+        
+        let lastAction = stringDiff(a: query, b:self.lastQuery)
+        let now =  NSDate().timeIntervalSince1970
+        let keydownData: [String: Any] = [
+            "event": "search.keystroke",
+            "created": now,
+            "sessionIdentifier":sessionid,
+            "queryString": query,
+            "lastAction": lastAction,
+            "userAgent": userAgent
+        ]
+        self.mapboxEvent(data: keydownData)
+        self.lastQuery =  query;
+    }
+    
+    /**
+     Check if reverse geocode
+    */
+    func isReverseGeocode(testStr:String) -> Bool {
+        print(testStr)
+        let reverseGeocodeRegex = "\\S+\\d+,\\S+\\d+"
+        
+        let test = NSPredicate(format:"SELF MATCHES %@", reverseGeocodeRegex)
+        return test.evaluate(with: testStr)
```

Comment:
A more straightforward alternative for regular expression matching would be NSRegularExpression or `NSString.replacingOccurrences(of:with:)` with the `NSString.CompareOptions.regularExpression` option.